storage/innobase/handler/handler0alter.cc

```diff
@@ -8430,6 +8430,16 @@ ha_innobase::prepare_inplace_alter_table(
 			DBUG_RETURN(true);
 		}
 
+		if ((ha_alter_info->handler_flags & ALTER_OPTIONS)
+		    && ctx->page_compression_level
+		    && !ctx->old_table->not_redundant()) {
+			my_error(ER_ILLEGAL_HA_CREATE_OPTION, MYF(0),
+				 table_type(),
+				 "PAGE_COMPRESSED in REDUNDANT ROW "
+				 "FORMAT");
```

Comment:
I would suggest a straight SQL snippet that works better with non-English error messages:
`"PAGE_COMPRESSED=1 ROW_FORMAT=REDUNDANT"`.